api/v1alpha1/s3instance_types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// S3InstanceSpec defines the desired state of S3Instance
+type S3InstanceSpec struct {
+
+	// type of the S3Instance
+	// +kubebuilder:validation:Required
+	S3Provider string `json:"s3Provider"`
+
+	// url of the S3Instance
+	// +kubebuilder:validation:Required
+	UrlEndpoint string `json:"urlEndpoint"`
+
+	// SecretName associated to the S3Instance containing accessKey and secretKey
+	// +kubebuilder:validation:Required
+	SecretName string `json:"secretName"`
+
+	// region associated to the S3Instance
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// useSSL when connecting to the S3Instance
+	// +kubebuilder:validation:Optional
+	UseSSL bool `json:"useSSL,omitempty"`
```

Comment:
Couldn't we get this info from URL?

https -> useSSL=true
http -> useSSL=false